src/main/java/org/opensearch/ad/ml/EntityColdStarter.java

```diff
@@ -329,85 +330,41 @@ private void trainModelFromDataSegments(List<double[][]> dataPoints, Entity enti
         }
 
         int rcfNumFeatures = dataPoints.get(0)[0].length;
-        RandomCutForest rcf = RandomCutForest
-            .builder()
-            .dimensions(rcfNumFeatures)
-            .sampleSize(rcfSampleSize)
-            .numberOfTrees(numberOfTrees)
-            .timeDecay(rcfTimeDecay)
-            .outputAfter(numMinSamples)
-            .parallelExecutionEnabled(false)
-            .compact(true)
-            .precision(Precision.FLOAT_32)
-            .boundingBoxCacheFraction(AnomalyDetectorSettings.REAL_TIME_BOUNDING_BOX_CACHE_RATIO)
-            // same with dimension for opportunistic memory saving
-            // Usually, we use it as shingleSize(dimension). When a new point comes in, we will
-            // look at the point store if there is any overlapping. Say the previously-stored
-            // vector is x1, x2, x3, x4, now we add x3, x4, x5, x6. RCF will recognize
-            // overlapping x3, x4, and only store x5, x6.
-            .shingleSize(shingleSize)
-            .build();
-        List<double[]> allScores = new ArrayList<>();
-        int totalLength = 0;
-        // get continuous data points and send for training
-        for (double[][] continuousDataPoints : dataPoints) {
-            double[] scores = trainRCFModel(continuousDataPoints, rcf);
-            allScores.add(scores);
-            totalLength += scores.length;
-        }
+        ExtendedRandomCutForest ercf = new ExtendedRandomCutForest(
+            RandomCutForest
```

Comment:
Why pass `RandomCutForest.builder` into `ExtendedRandomCutForest` rather than `RandomCutForest` ?

Reply:
Because that's the current api available. There is no reason another constructor with a rcf cannot be added. When it's added, the client code here can change to use that.